src/texture.rs

```diff
@@ -80,8 +80,8 @@ impl Texture {
     }
 
     /// Access resolution and adjacency information about a face.
-    pub fn face_info(&self, face_id: i32) -> &FaceInfo {
-        unsafe { sys::ptextexture_get_face_info(self.0, face_id) }
+    pub fn face_info(&self, face_id: i32) -> FaceInfo {
+        FaceInfo(*unsafe { sys::ptextexture_get_face_info(self.0, face_id) })
```

Comment:
One thing I find a bit confusing about this change in particular, is the functions like `set_resolution(&mut self)`
IIUC these calls to e.g. `set_resolution` are affecting a locally owned copy of the `FaceInfo`, rather than setting the resolution in a way that affects the texture's face_info.

It seems worth asking if this situation is as intended.